orderbook/src/main.rs

```diff
@@ -214,11 +214,19 @@ async fn main() {
             .unwrap();
 
     let pool_aggregator = PoolAggregator::from_providers(&pair_providers, &web3).await;
-    let pool_fetcher =
-        CachedPoolFetcher::new(Box::new(pool_aggregator), current_block_stream.clone());
+    let pool_fetcher = Arc::new(
+        PoolCache::new(
+            args.shared.pool_cache_blocks,
+            args.shared.pool_cache_lru_size,
+            args.shared.pool_cache_maximum_recent_block_age,
+            Box::new(pool_aggregator),
+            current_block_stream.clone(),
```

Comment:
Is it an issue that we have separate instances of `current_block_stream` where the "current block" might be different for them because of polling offsets?

For example the `maintenance` block stream could see a new block, but the polling is offset in such a way that the `pool_cache` block stream doesn't see it, and then it will, instead, update for the previous block instead of the next one.

**edit**: I see we are using tokio spmc channels (instead of mpsc) so it is meant for a single producer (the spawned fiber) an multiple receivers, so its not an issue at all.